### PR TITLE
Fixes #215: Parameters can be saved without a value

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

**Comment by domcleal** [comment]
Other models (host groups, hosts, domains, taxonomies) have the same use of nested attributes, should they be changed?

**Comment by orrabin** [comment]
@domcleal it can be confusing, everything looks like it was saved and only when you look for your parameter you notice it isn't there. 
I'm not sure anyone would actually try to create a parameter without a value but I can change the rest or the models or add a warning at the top of each parameter form that parameters without values will be rejected.
What do you think?

**Comment by orrabin** [comment]
All models with parameters as nested attributes now act the same way as os.

**Comment by domcleal** [comment]
@orrabin looks good and tests well.  Could you please fix the test failures and rebase?  Thanks.

Nitpick: you can remove the "cancels #6697" from the commit message now, that's fine.

**Comment by orrabin** [comment]
@domcleal I removed the tests that weren't relevant anymore

File: test/unit/common_parameter_test.rb [hunk 1]
```diff
@@ -18,18 +18,6 @@ class CommonParameterTest < ActiveSupport::TestCase
     assert parameter.save
   end
 
-  test "value can't be blank" do
```
Comment (domcleal) [inline test/unit/common_parameter_test.rb:21]: I think it'd be better to invert the logic of the test: just have it check that the value _can_ be blank, and can be empty.

Comment (orrabin) [inline test/unit/common_parameter_test.rb:21]: @domcleal thanks, I changed the tests

**Comment by domcleal** [comment]
Thanks @orrabin, that's the oldest bug squashed in a long time :)

Merged as 32ced38 for Foreman 1.6.0.

**Pull request closed without merging**